lib/autoupdate/tools/updater.go

```diff
@@ -114,38 +114,44 @@ func NewUpdater(tools []string, toolsDir string, localVersion string, options ..
 // CheckLocal is run at client tool startup and will only perform local checks.
 // Returns the version needs to be updated and re-executed, by re-execution flag we
 // understand that update and re-execute is required.
-func (u *Updater) CheckLocal() (version string, reExec bool) {
+func (u *Updater) CheckLocal() (version string, reExec bool, err error) {
 	// Check if the user has requested a specific version of client tools.
 	requestedVersion := os.Getenv(teleportToolsVersionEnv)
 	switch requestedVersion {
 	// The user has turned off any form of automatic updates.
 	case "off":
-		return "", false
+		return "", false, nil
 	// Requested version already the same as client version.
 	case u.localVersion:
-		return u.localVersion, false
+		return u.localVersion, false, nil
+	}
+
+	if requestedVersion != "" {
+		if _, err := semver.NewVersion(requestedVersion); err != nil {
+			return "", false, trace.WrapWithMessage(err, "not valid semantic version")
+		}
+		return requestedVersion, true, nil
```

Comment:
nit: would it make sense to factor this into the `switch` above or is that too unclear? e.g.

```go
switch requestedVersion {
// The user has turned off any form of automatic updates.
case "off":
	return "", false, nil
// Requested version already the same as client version.
case u.localVersion:
	return u.localVersion, false, nil
// No requested version, we continue
case "":
// Requested version that is not the local one
default:
	if _, err := semver.NewVersion(requestedVersion); err != nil {
		return "", false, trace.WrapWithMessage(err, "not valid semantic version")
	}
	return requestedVersion, true, nil
}
```